x-pack/plugins/apm/server/routes/latency_distribution/route.ts

```diff
@@ -61,9 +62,18 @@ const latencyOverallTransactionDistributionRoute = createApmServerRoute({
       termFilters,
     } = resources.params.body;
 
+    const searchAggregatedTransactions = await getSearchAggregatedTransactions({
+      ...setup,
+      kuery,
+      start,
+      end,
+    });
+
     return getOverallLatencyDistribution({
       setup,
-      eventType: ProcessorEvent.transaction,
+      eventType: searchAggregatedTransactions
```

Comment:
I'm not sure if this is enough. I think you also need a filter for `{ exists: { field: 'transaction.duration.histogram' } }`. We have a helper function for that purpose.

Reply:
The call to `getSearchAggregatedTransactions` on line 65 does call the helper [`getHasAggregatedTransactions`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/server/lib/helpers/transactions/index.ts#L40=) which filters for this field.

Reply:
It only checks for the existence. The actual search itself needs a filter on `{ exists: { field: 'transaction.duration.histogram' } }` otherwise it will run over all metric documents.